tests/chainerx_tests/unit_tests/routines_tests/test_logic.py

```diff
@@ -205,3 +209,98 @@ def forward_xp(self, inputs, xp):
         a, = inputs
         b = xp.logical_not(a)
         return b,
+
+
+def compute_all(xp, a, axis, keepdims):
+    return xp.all(a, axis=axis, keepdims=keepdims)
+
+
+def compute_any(xp, a, axis, keepdims):
+    return xp.any(a, axis=axis, keepdims=keepdims)
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'shape,axis': [
+            ((), None),
+            ((), ()),
+            ((2,), None),
+            ((2,), ()),
+            ((2,), 0),
+            ((2,), (0,)),
+            ((2,), (-1,)),
+            ((2, 3), None),
+            ((2, 3), ()),
+            ((2, 3), 0),
+            ((2, 3), (0,)),
+            ((2, 3), (1,)),
+            ((2, 3), (-1,)),
+            ((2, 3), (-2,)),
+            ((2, 3), (0, 1)),
+            ((2, 3), (-2, -1)),
+            ((1, 3), None),  # Reduce over 1-dim axis
+            ((0, 3), None),  # Reduce over 0-dim axis
+            # Reduce over axes that are in the middle or apart
+            ((2, 3, 4), (1,)),
+            ((2, 3, 4), (0, 2)),
+            # Reduce over axes that are apart and/or unsorted
+            ((2, 3), (1, 0)),
+            ((2, 3, 4), (2, 0)),
+            ((2, 3, 4), (2, 0, 1)),
+            ((2, 3, 4), (-2, 2, 0)),
+        ],
+        'keepdims': [True, False],
+        'in_dtypes':
+            _make_in_dtypes(1, chainerx.testing.numeric_dtypes),
```

Comment:
`_make_in_dtypes` takes only `1` as the argument of `number_of_in_params` in this file.
How about writing `'dtype': chainerx.testing.all_dtypes` here?

Reply:
Can't directly use `chainerx.testing.all_dtypes` as it is a `Tuple`. `chainer.testing.product` expects `List of Tuple`. Hence using `_make_in_dtypes`.

Using `chainerx.testing.all_dtypes`.

Sounds ok?

Reply:
How about fixing L363?

Reply:
Sorry, `in_dtype` is better.

Reply:
Sure, will change the name to `in_dtype`.